### PR TITLE
java Thread task

### PR DESCRIPTION
**Pull request by Truput**
java Thread task added

### CODE REVIEW

File: src/com/epam/java/javathread/threads/CarParkingThread.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            System.out.println(parkingQueue);
+            if (parkingQueue.offer(car, 6, TimeUnit.SECONDS)){
+                car.park();
+                System.out.println(car.getMake() + " is parked and staying");
+                TimeUnit.SECONDS.sleep(new Random().nextInt(10) + 1);
+                System.out.println(car.getMake() + " leaves parking place");
+                car.move();
+                parkingQueue.remove(car);
+                System.out.println(parkingQueue);
+            } else {
```
Comment (voljabes) [inline src/com/epam/java/javathread/threads/CarParkingThread.java:36]: Consider to make the same spaces before else and catch and use the similar approach everywhere

File: src/com/epam/java/javathread/threads/CarParkingThread.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    public void run() {
+        System.out.println(car.getMake() + " is entering the parking");
+        car.move();
+        try {
+            System.out.println(car.getMake() + " is trying to park");
+            System.out.println(parkingQueue);
+            if (parkingQueue.offer(car, 6, TimeUnit.SECONDS)){
```
Comment (voljabes) [inline src/com/epam/java/javathread/threads/CarParkingThread.java:28]: Please consider to introduce a local variable for TimeUnit.SECONDS if it is used more than 1 time

File: src/com/epam/java/javathread/regexp/RegexpMatcher.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.epam.java.javathread.regexp;
+
+public class RegexpMatcher {
+    private final static String STRING_REGEX = "[a-zA-Z]+";
+
+    public static boolean isMatch(String string){
+        return string.matches(STRING_REGEX);
+    }
+
```
Comment (voljabes) [inline src/com/epam/java/javathread/regexp/RegexpMatcher.java:9]: Consider to remove an extra empty line

File: src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    public double getPARKING_SPACE() {
+        return PARKING_SPACE;
+    }
+
+    public boolean isOccupied() {
+        return isOccupied;
+    }
+
+    public void setIsOccupied(boolean occupied) {
```
Comment (voljabes) [inline src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java:19]: Consider to rename it to just setOccupied

File: src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+public class ParkingPlace {
+    private final double PARKING_SPACE = 10.0;
+    private boolean isOccupied;
+
+    public ParkingPlace(boolean isOccupied) {
+        this.isOccupied = isOccupied;
+    }
+
+    public double getPARKING_SPACE() {
```
Comment (voljabes) [inline src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java:11]: Please consider to rename method to getParkingSpace

File: src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package com.epam.java.javathread.maintask.parking.parkingplace;
+
+public class ParkingPlace {
+    private final double PARKING_SPACE = 10.0;
```
Comment (voljabes) [inline src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java:4]: Does this variable mean amount of parking places?

File: src/com/epam/java/javathread/maintask/parking/Parking.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    public int getNumberOfParkingPlaces() {
+        return numberOfParkingPlaces;
+    }
+
+    public ParkingPlace[] getParkingPlaces() {
+        return parkingPlaces;
+    }
+
+    public void parkCar(Car car){
+
```
Comment (voljabes) [inline src/com/epam/java/javathread/maintask/parking/Parking.java:67]: Consider to remove an extra empty line

File: src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java (hunk 6 above)
Comment (Truput) [reply src/com/epam/java/javathread/maintask/parking/parkingplace/ParkingPlace.java:4]: no, it means that parking space's square is 10 m^2

**Review by voljabes — APPROVED** [review]

**Pull request merged**